diff --git a/examples/fremantle/QmlComponentGallery/qml/VisibilityPage.qml b/examples/fremantle/QmlComponentGallery/qml/VisibilityPage.qml
index ba3770f..ee02ec5 100644
--- a/examples/fremantle/QmlComponentGallery/qml/VisibilityPage.qml
+++ b/examples/fremantle/QmlComponentGallery/qml/VisibilityPage.qml
@@ -106,43 +106,36 @@ Page {
         }
     }
     
-    Flickable {
-        id: flickable
+    Row {
+        id: row
+        spacing: 10
         anchors.fill: parent
-        contentWidth: col.width
-        contentHeight: col.height
-        flickableDirection: Flickable.VerticalFlick    
-
-        Column {
-            id: col
-            spacing: 10
-            width: flickable.width
-            
-            Loader {
-                id: l1
-                sourceComponent: textBox
-            }
-    
-            Loader {
-                id: l2
-                sourceComponent: textBox
-            }
-    
-            Loader {
-                id: l3
-                sourceComponent: textBox
-            }
-    
-            Component.onCompleted: {
-                updateViewMode();
-                updateVisible();
-                updateActive();
-                
-                var count = children.length;
-                for (var i = 0; i < count; i++) {
-                    var item = children[i];
-                    item.anchors.horizontalCenter = item.parent.horizontalCenter;
-                }                
+        anchors.leftMargin: (parent.width - childrenRect.width) / 2
+
+        Loader {
+            id: l1
+            sourceComponent: textBox
+        }
+
+        Loader {
+            id: l2
+            sourceComponent: textBox
+        }
+
+        Loader {
+            id: l3
+            sourceComponent: textBox
+        }
+
+        Component.onCompleted: {
+            updateViewMode();
+            updateVisible();
+            updateActive();
+
+            var count = children.length;
+            for (var i = 0; i < count; i++) {
+                var item = children[i];
+                item.anchors.verticalCenter = item.parent.verticalCenter;
             }
         }
     }
