diff --git a/src/meego/plugin.cpp b/src/meego/plugin.cpp
index 3018247..b925e57 100644
--- a/src/meego/plugin.cpp
+++ b/src/meego/plugin.cpp
@@ -77,6 +77,7 @@
 
 #include "i18n/mlocalewrapper.h"
 
+static const char *VERSION_FLAVOR = "fremantle-cssu";
 static const int   VERSION_MAJOR  = 1;
 static const int   VERSION_MINOR  = 0;
 
@@ -102,6 +103,7 @@ public:
             engine->rootContext()->setContextProperty("screen", MDeclarativeScreen::instance());
             qmlRegisterUncreatableType<MDeclarativeScreen>(uri, 1, 0, "Screen", "");
 
+ 	    engine->rootContext()->setContextProperty("version_flavour", VERSION_FLAVOR);
             engine->rootContext()->setContextProperty("version_major", VERSION_MAJOR);
             engine->rootContext()->setContextProperty("version_minor", VERSION_MINOR);
 
