diff --git a/src/meego/mdeclarativeinputcontext.h b/src/meego/mdeclarativeinputcontext.h
index bb33023..929aa38 100644
--- a/src/meego/mdeclarativeinputcontext.h
+++ b/src/meego/mdeclarativeinputcontext.h
@@ -56,6 +56,7 @@ public:
     MDeclarativeInputContext(QDeclarativeItem *parent = 0);
     virtual ~MDeclarativeInputContext();
 
+    Q_PROPERTY(bool simulateSip READ simulateSip FINAL);
     Q_PROPERTY(bool softwareInputPanelVisible READ softwareInputPanelVisible NOTIFY softwareInputPanelVisibleChanged FINAL)
     Q_PROPERTY(QRect softwareInputPanelRect READ softwareInputPanelRect NOTIFY softwareInputPanelRectChanged FINAL)
     Q_PROPERTY(QRectF microFocus READ microFocus NOTIFY microFocusChanged) //TODO: will be removed on w13 - remove "NOTIFY microFocusChanged"
@@ -77,6 +78,8 @@ public:
 
     Q_INVOKABLE void clearClipboard();
 
+    bool simulateSip() const;
+
     bool softwareInputPanelVisible() const;
     QRect softwareInputPanelRect() const;
 
