diff --git a/src/fremantle/Sheet.qml b/src/fremantle/Sheet.qml
index e933068..f4c6f84 100644
--- b/src/fremantle/Sheet.qml
+++ a/src/fremantle/Sheet.qml
@@ -114,16 +114,8 @@
     Item {
         id: sheet
 
-        //when the sheet is part of a page do nothing
-        //when the sheet is a direct child of a PageStackWindow, consider the status bar
-        property int statusBarOffset: (typeof __isPage != "undefined") ? 0
-                                     : (typeof __statusBarHeight == "undefined") ? 0
-                                     :  __statusBarHeight
-        
         width: parent.width
-        height: parent.height - statusBarOffset
-
-        y: statusBarOffset
+        height: parent.height
 
         clip: true
         
