diff --git a/src/fremantle/style/PageStackWindowStyle.qml b/src/fremantle/style/PageStackWindowStyle.qml
index 64e7c54..2797b72 100644
--- b/src/fremantle/style/PageStackWindowStyle.qml
+++ a/src/fremantle/style/PageStackWindowStyle.qml
@@ -39,8 +39,9 @@
 ****************************************************************************/
 
 import QtQuick 1.1
+import "." 1.0
 
-Style {
+WindowStyle {
     property bool cornersVisible: true
 
     // Background
