diff --git a/src/meego/Label.qml b/src/meego/Label.qml
index f83312a..063f2be 100644
--- a/src/meego/Label.qml
+++ b/src/meego/Label.qml
@@ -50,9 +50,19 @@ Text {
     //Deprecated, TODO Remove this on w13
     property alias style: root.platformStyle
 
-    font.family: platformStyle.fontFamily
-    font.pixelSize: platformStyle.fontPixelSize
+    font.family: ""
+    font.pixelSize: 0
+    
     color: platformStyle.textColor
 
     wrapMode: Text.Wrap
+
+    Component.onCompleted: {
+        if (font.family === "") {
+            font.family = platformStyle.fontFamily
+        }
+        if (font.pixelSize === 0) {
+            font.pixelSize = platformStyle.fontPixelSize
+        }
+    }
 }
