--- a/src/fremantle/plugin.cpp	2011-12-08 13:45:18.499635102 +0100
+++ b/src/fremantle/plugin.cpp	2011-12-08 13:45:18.499635102 +0100
@@ -75,6 +75,7 @@
 
 #include "i18n/mlocalewrapper.h"
 
+static const char *VERSION_FLAVOR = "fremantle-ssu";
 static const int   VERSION_MAJOR  = 1;
 static const int   VERSION_MINOR  = 0;
 
@@ -84,12 +85,7 @@
 
 public:
     void initializeEngine(QDeclarativeEngine *engine, const char *uri) {
-        Q_ASSERT(uri == QLatin1String("com.meego") || uri == QLatin1String("com.nokia.meego") || uri == QLatin1String("Qt.labs.components.native"));
-
-        if(uri == QLatin1String("com.meego")) {
-            qWarning() << "'import com.meego' is DEPRECATED and may cause 'Error: Cannot assign QObject* to PageStack_QMLTYPE_42*', missing icons, widgets and other problems;"
-                          "use 'import com.nokia.meego' instead";
-        }
+        Q_ASSERT(uri == QLatin1String("org.maemo.fremantle") || uri == QLatin1String("Qt.labs.components.native"));
 
         QDeclarativeExtensionPlugin::initializeEngine(engine, uri);
 
@@ -99,7 +95,8 @@
 
             engine->rootContext()->setContextProperty("screen", MDeclarativeScreen::instance());
             qmlRegisterUncreatableType<MDeclarativeScreen>(uri, 1, 0, "Screen", "");
-
+ 
+	    engine->rootContext()->setContextProperty("version_flavour", VERSION_FLAVOR);
 	    engine->rootContext()->setContextProperty("version_major", VERSION_MAJOR);
             engine->rootContext()->setContextProperty("version_minor", VERSION_MINOR);
 
@@ -139,7 +136,7 @@
     }
 
     void registerTypes(const char *uri) {
-        Q_ASSERT(uri == QLatin1String("com.meego") || uri == QLatin1String("com.nokia.meego") || uri == QLatin1String("Qt.labs.components.native"));
+        Q_ASSERT(uri == QLatin1String("org.maemo.fremantle") || uri == QLatin1String("Qt.labs.components.native"));
         qmlRegisterType<MSnapshot>(uri, 1, 0, "Snapshot");
 
         qmlRegisterUncreatableType<MPageStatus>(uri, 1, 0, "PageStatus", "");
@@ -152,6 +149,7 @@
         // Custom primitives
         qmlRegisterType<MDeclarativeImplicitSizeItem>(uri, 1, 0, "ImplicitSizeItem");
         qmlRegisterType<MDeclarativeMaskedItem>(uri, 1, 0, "MaskedItem");
+        qmlRegisterType<MDeclarativeFocusScope>(uri, 1, 0, "FocusScope");
         qmlRegisterType<MInverseMouseArea>(uri, 1, 0, "InverseMouseArea");
         qmlRegisterType<MDeclarativeMouseFilter>(uri, 1, 0, "MouseFilter");
         qmlRegisterType<MDeclarativeMouseEvent>(uri, 1, 0, "MMouseEvent");
